src/evaluatorfrontend.cpp

```diff
@@ -100,41 +103,54 @@ void EvaluatorFrontend::RunRTEvaluator() {
 	auto endTime = std::chrono::steady_clock::now();
 	bool remainingStatesNotAdded = true;
 
+	boost::lockfree::queue<NetworkState *> stateQueue(1024);
+
+	if (!stateQueue.is_lock_free()) {
+		throw std::runtime_error("Queue is not lockfree");
+	}
+
+	std::thread threadObj([this, &stateQueue] {
+		while (!evaluator->IsFinished()) {
+			states.push_back(evaluator->GetNextNetworkState());
+			if (printStd)
+				std::cout << states.back().PrintCsvRow();
+			NetworkState *copy = new NetworkState(states.back());
+			if (!stateQueue.push(copy)) {
+				throw std::runtime_error("Push failed");
+			}
+		}
+	});
+
 	while (!mPlot->RunPlot()) {
-		if (EvaluatorFunc()) {
-			endTime = std::chrono::steady_clock::now();
-			auto elapsedTime = std::chrono::duration_cast<std::chrono::milliseconds>(
-					endTime - startTime);
-			auto updateTime = std::chrono::milliseconds(UpdateRate);
+		endTime = std::chrono::steady_clock::now();
+		auto elapsedTime = std::chrono::duration_cast<std::chrono::milliseconds>(
+				endTime - startTime);
+		auto updateTime = std::chrono::milliseconds(UpdateRate);
 
-			if (elapsedTime > updateTime) {
-				startTime = std::chrono::steady_clock::now();
+		if (elapsedTime > updateTime) {
+			startTime = std::chrono::steady_clock::now();
 
-				for (int i = 0; i < toPlot.size(); i++) {
-					int currentSpeicePointSize = toPlot[i].Function.size();
-					for (int j = currentSpeicePointSize; j < states.size(); j++) {
-						auto state = states[j];
-						toPlot[i].Function.push_back(
-								(OpenRTP::Point){static_cast<float>(state.time),
-																 static_cast<float>(state[toPlot[i].Name])});
-					}
-				}
+			while (!stateQueue.empty()) {
+				NetworkState *state = nullptr;
+				bool res = stateQueue.pop(state);
+				if (!res)
+					throw std::runtime_error("Could not pop from state queue");
 
-				mPlot->UpdatePlot();
-			}
-		} else if (remainingStatesNotAdded) {
-			for (int i = 0; i < toPlot.size(); i++) {
-				int currentSpeicePointSize = toPlot[i].Function.size();
-				for (int j = currentSpeicePointSize; j < states.size(); j++) {
-					toPlot[i].Function.push_back(
-							(OpenRTP::Point){static_cast<float>(states[j].time),
-															 static_cast<float>(states[j][toPlot[i].Name])});
+				for (int i = 0; i < toPlot.size(); i++) {
+					auto &name = toPlot[i].Name;
+					auto &time = state->time;
+					double value = 0;
+					value = state->at(toPlot[i].Name);
+					toPlot[i].Function.push_back((OpenRTP::Point){
+							static_cast<float>(time), static_cast<float>(value)});
 				}
+				delete state;
 			}
+
 			mPlot->UpdatePlot();
```

Comment:
Since EvaluatorFunc() is removed, the mPlot->UpdatePlot() will always be executed about 62.5 times per second, when it actually only should be run each time toPlot is updated.